packages/jest-worker/src/workers/threadChild.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import {
+  CHILD_MESSAGE_CALL,
+  CHILD_MESSAGE_END,
+  CHILD_MESSAGE_INITIALIZE,
+  PARENT_MESSAGE_ERROR,
+  PARENT_MESSAGE_OK,
+} from '../types';
+
+import type {ChildMessageInitialize, ChildMessageCall} from '../types';
+
+let file = null;
+
+/* eslint-disable import/no-unresolved */
+// $FlowFixMe: Flow doesn't support experimental node modules
+import {parentPort, isMainThread} from 'worker_threads';
+/* eslint-enable import/no-unresolved */
+
+/**
+ * This file is a small bootstrapper for workers. It sets up the communication
+ * between the worker and the parent process, interpreting parent messages and
+ * sending results back.
+ *
+ * The file loaded will be lazily initialized the first time any of the workers
+ * is called. This is done for optimal performance: if the farm is initialized,
+ * but no call is made to it, child Node processes will be consuming the least
+ * possible amount of memory.
+ *
+ * If an invalid message is detected, the child will exit (by throwing) with a
+ * non-zero exit code.
+ */
+parentPort.on(
+  'message',
+  (request: any /* Should be ChildMessage */): void => {
+    switch (request[0]) {
+      case CHILD_MESSAGE_INITIALIZE:
+        const init: ChildMessageInitialize = request;
+        file = init[2];
+        break;
+
+      case CHILD_MESSAGE_CALL:
+        const call: ChildMessageCall = request;
+        execMethod(call[2], call[3]);
+        break;
+
+      case CHILD_MESSAGE_END:
+        process.exit(0);
+        break;
+
+      default:
+        throw new TypeError(
+          'Unexpected request from parent process: ' + request[0],
+        );
+    }
+  },
+);
+
+function reportSuccess(result: any) {
+  if (isMainThread) {
+    throw new Error('Child can only be used on a forked process');
+  }
+
+  parentPort.postMessage([PARENT_MESSAGE_OK, result]);
+}
+
+function reportError(error: Error) {
+  if (isMainThread) {
+    throw new Error('Child can only be used on a forked process');
+  }
+
+  if (error == null) {
+    error = new Error('"null" or "undefined" thrown');
+  }
+
+  parentPort.postMessage([
+    PARENT_MESSAGE_ERROR,
+    error.constructor && error.constructor.name,
+    error.message,
+    error.stack,
+    // $FlowFixMe: this is safe to just inherit from Object.
+    typeof error === 'object' ? Object.assign({}, error) : error,
+  ]);
+}
+
+function execMethod(method: string, args: $ReadOnlyArray<any>): void {
```

Comment:
Can you create a common file within the `workers/` folder that abstract this functionality? It looks like `child` and `threadChild` are 90% the same.